config/config/src/index.ts

```diff
@@ -637,3 +639,44 @@ function parsePackageManager (packageManager: string): { name: string, version:
     version,
   }
 }
+
+function addSettingsFromWorkspaceManifestToConfig (pnpmConfig: Config, {
+  acceptNonRc,
+  configFromCliOpts,
+  projectManifest,
+  workspaceManifest,
+  workspaceDir,
+}: {
+  acceptNonRc: boolean
+  configFromCliOpts: Record<string, unknown>
+  projectManifest: ProjectManifest | undefined
+  workspaceDir: string | undefined
+  workspaceManifest: WorkspaceManifest
+}): void {
+  const newSettings = Object.assign(getOptionsFromPnpmSettings(workspaceDir, workspaceManifest, projectManifest), configFromCliOpts)
+  for (const [key, value] of Object.entries(newSettings)) {
+    if (!isCamelCase(key)) continue
+
+    // @ts-expect-error
+    pnpmConfig[key] = value
+
+    const kebabKey = kebabCase(key)
+    // Q: Why `types` instead of `rcOptionTypes`?
+    // A: `rcOptionTypes` includes options that would matter to the `npm` cli which wouldn't care about `pnpm-workspace.yaml`.
+    const isRc = kebabKey in types
```

Comment:
I am not sure all settings from types should be global. For instance, the `sync-injected-deps-after-scripts` probably doesn't make sense to be global.